stac_api/infrastructure/construct.py

```diff
@@ -70,7 +70,11 @@ def __init__(
             lambda_function, port_range=aws_ec2.Port.tcp(5432)
         )
 
-        lambda_function.add_environment("TITILER_ENDPOINT", raster_api.raster_api.url)
+        if veda_stac_settings.custom_host:
+            titler_endpoint = f"https://{veda_stac_settings.custom_host}/api/raster/"
+        else:
+            titler_endpoint = raster_api.raster_api.url
+        lambda_function.add_environment("TITILER_ENDPOINT", titler_endpoint)
```

Comment:
```suggestion
        if veda_stac_settings.custom_host:
            titiler_endpoint = f"https://{veda_stac_settings.custom_host}/api/raster/"
        else:
            titiler_endpoint = raster_api.raster_api.url
        lambda_function.add_environment("TITILER_ENDPOINT", titiler_endpoint)
```

nit: we could also pull the raster root path into `veda_stac_settings` and use it to construct more of the URL here